share/functions/__fish_whatis_current_token.fish

```diff
@@ -1,11 +1,25 @@
 # This function is typically bound to Alt-W, it is used to list man page entries
 # for the command under the cursor.
-function __fish_whatis_current_token -d "Show man page entries related to the token under the cursor"
-    set -l tok (commandline -pt)
+function __fish_whatis_current_token -d "Show man page entries or function description related to the token under the cursor"
+    set -l token (commandline -pt)
 
-    if test -n "$tok[1]"
+    if test -n "$token"
         printf "\n"
-        whatis $tok[1]
+        whatis $token 2>/dev/null
+
+        if test $status -eq 16 # no match found
```

Comment:
The value 16 here is a Linux-ism. Looks like macOS and BSD return success even if nothing was found :( I'm not sure what to do - maybe just leave it.

Reply:
I changed the code so that commands have least precedence when searching for a description and it doesn't depend on`whatis`'s exit status.